app/bundles/CoreBundle/Helper/ExportHelper.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/*
+ * @copyright   2020 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\CoreBundle\Helper;
+
+use ArrayIterator;
+use Iterator;
+use PhpOffice\PhpSpreadsheet\IOFactory;
+use PhpOffice\PhpSpreadsheet\Spreadsheet;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Provides several functions for export-related tasks,
+ * like exporting to CSV or Excel.
+ */
+class ExportHelper
```

Comment:
This is just a note how it could be done using OOP. No need to refactor. Your implementation will work just fine. Just FYI:
```php
interface ExporterInterface
{
   public function export(Iterator $data, string $name): StreamedResponse;
}
```
The classes for Excel, CSV, ... exporters would implement the `export` method:
```php
final class ExcelExporter implements ExporterInterface
{
    public const TYPE = 'xlsx';
    public function export(Iterator $data, string $name): StreamedResponse
    { ... }
}
```
The simple option how to instantiate the exporters would be a Factory. Or they could be created by the event dispatcher or Symfony's DI to make it more extensible but I think that would be overkill for this use case.
```php
class ExporterFactory
{
    public function make(string $type): ExporterInterface
    {
        switch ($type) {
            case CsvExporter::TYPE:
                return new CsvExporter();
            case ExcelExporter::TYPE:
                return new ExcelExporter();
            default: 
                throw new \Exception("There is no exporter for {$type}");
    }
}
```
Usage in a controller:
```php
return $exporterFactory->make($type)->export($data, $name);
```

Reply:
Thanks!! Will keep that in mind for the future 😊